module/Photo/src/Photo/Service/Photo.php

```diff
@@ -81,13 +80,87 @@ public function storeUploadedPhoto($path, $target_album)
      * @param \Photo\Model\Photo $photo the photo to add the metadata to
      * @return \Photo\Model\Photo the photo with the added metadata
      */
-    protected function populateMetadata($photo)
+    protected function populateMetadata($photo,$path)
     {
-        //TODO: fetch metadata from photo and add it
-        //placeholder for now
-        $photo->setDate(new \DateTime("2014-12-12 12:13:12.424242"));
+        $exif = \read_exif_data($path,'EXIF');
+         if (isset($exif['Artist'])){
+            $photo->setArtist($exif['Artist']);
+        }
+        else{
+            $photo->setArtist("Unknown");//Needs to be t9n'd in the view
+        }
+        //I assume the exif data isn't deliberately stripped, so most values 
+        //are assumed to exist.
+        $photo->setCamera($exif['Model']);
+        $photo->setDateTime(\date_create($exif['DateTimeOriginal']));
+        $photo->setFlash($exif['Flash']==0);
+        $photo->setFocalLength($this->frac2dec($exif['FocalLength']));
+        $photo->setExposureTime($this->frac2dec($exif['ExposureTime']));
+        $photo->setShutterSpeed($this->exifGetShutter($exif));
+        $photo->setAperture($this->exifGetFstop($exif));
+        $photo->setIso($exif['ISOSpeedRatings']);
         return $photo;
     }
+    
+    /*
+     * NOTE: Most code in the following part is copied from 
+     * the old site, mostly because I lack knowledge in photography.
+     */
+    
+    /**
+     * Convert a string representing a rational number to a string representing 
+     * the corresponding decimal approximation. 
+     * @param string $str the rational number, represented as num+'/'+den
+     * @return float the decimal number, represented as float
+     */
+    private function frac2dec($str) 
+    {
+        list($n, $d) = \explode('/', $str);
+        //Old site suppressed errors of previous line. No clue why.
+        if (!empty($d)) {
+            return $n / $d;
+        }
+        return $str;
+    }
+    
+    /**
+     * Computes the shutter speed from the exif data.
+     * @param array $exif the exif data extracted from the photo.
+     * @return string the shutter speed, represented as a rational string.
+     */
+    private function exifGetShutter(&$exif) 
```

Comment:
Het is niet nodig om hier de & te gebruiken. Pass-by-reference maakt dingen meestal alleen maar onvoorspelbaar. Tenzij dat het expliciet nodig is, is het beter om dit niet te doen. En vaak is het ook beter om dit te refactoren op zo'n manier dat dat nog steeds niet nodig is.